components/crypto/src/crypto_lib/sodiumoxide/kx.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    fmt,
+    ops::{Index, Range, RangeFrom, RangeFull, RangeTo},
+};
+
+use crate::{write_short_hex, Seed};
+use exonum_sodiumoxide::crypto::{
+    kx,
+    scalarmult::curve25519::{
+        scalarmult_base as sodium_scalarmult_base, Scalar as Curve25519Scalar,
+    },
+};
+
+use hex::{encode as encode_hex, FromHex, FromHexError};
+
+use serde::{
+    de::{self, Deserialize, Deserializer, Visitor},
+    Serialize, Serializer,
+};
+
+pub fn gen_keypair() -> (PublicKey, SecretKey) {
+    let (pk, sk) = kx::gen_keypair();
+
+    (PublicKey(pk), SecretKey(sk))
+}
+
+pub fn gen_keypair_from_seed(seed: &Seed) -> (PublicKey, SecretKey) {
+    let (pk, sk) = kx::keypair_from_seed(&kx::Seed::from_slice(&seed[..]).unwrap());
+
+    (PublicKey(pk), SecretKey(sk))
+}
+
+#[derive(Debug, Default, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct KeyPair {
+    pub(crate) public_key: PublicKey,
+    pub(crate) secret_key: SecretKey,
+}
+
+impl KeyPair {
+    pub fn from_keys(public_key: PublicKey, secret_key: SecretKey) -> Self {
```

Comment:
I think that `From<(PublicKey, SecretKey>)` implementation can be very useful also.